packages/lesswrong/components/ea-forum/onboarding/EAOnboardingFlow.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Components, registerComponent } from "../../../lib/vulcan-lib";
+import { useCurrentUser } from "../../common/withUser";
+import { EAOnboardingContextProvider } from "./useEAOnboarding";
+
+const styles = (_theme: ThemeType) => ({
+  root: {
+    padding: 0,
+  },
+});
+
+const EAOnboardingFlow = ({classes}: {
+  classes: ClassesType<typeof styles>,
+}) => {
+  const currentUser = useCurrentUser();
+
+  // If `usernameUnset` is true, then we need to show the onboarding flow.
+  // We cache the value in a `useState` as it gets set to false in the very
+  // first stage (which is the only compulsary stage) - without caching the
+  // value this would close the popup.
+  const [isOnboarding, setIsOnboarding] = useState(currentUser?.usernameUnset);
+
+  useEffect(() => {
+    // Set `isOnboarding` to true after a new user signs up.
+    setIsOnboarding((currentValue) => currentValue || currentUser?.usernameUnset);
+  }, [currentUser?.usernameUnset]);
+
+  const onOnboardingComplete = useCallback(() => {
+    setIsOnboarding(false);
+  }, []);
+
+  if (!isOnboarding) {
+    return null;
+  }
+
+  const {
+    BlurredBackgroundModal, EAOnboardingUserStage, EAOnboardingSubscribeStage,
+    EAOnboardingWorkStage, EAOnboardingThankYouStage,
+  } = Components;
+  return (
+    <BlurredBackgroundModal open className={classes.root}>
+      <EAOnboardingContextProvider onOnboardingComplete={onOnboardingComplete}>
+        <EAOnboardingUserStage />
```

Comment:
We render _all_ the stages here (though all except one will return null). This allows us to do things like pre-loading the suggested tags and users to subscribe to before we get to that stage to make the flow a bit more seamless.